### PR TITLE
[2.0] SSH2: Add support for curve25519-sha256@libssh.org.txt via libsodium-php.

### LINKED ISSUE #610: Add curve25519-sha256@libssh.org key exchange algorithm
It seems that some server configurations are now upgraded to use only curve25519-sha256@libssh.org or some more advanced key exchange algorithms than the diffie-hellman-group1-sha1 or diffie-hellman-group14-sha1 currently supported by phpseclib.
Would it be possible to add curve25519-sha256@libssh.org ?

### PR DESCRIPTION
**Pull request by bantu**
Fix #610

See https://github.com/jedisct1/libsodium-php

### CODE REVIEW

**Comment by bantu** [comment]
Diff is easier to read without indentation changes: https://github.com/phpseclib/phpseclib/pull/755/files?w=0

File: phpseclib/Net/SSH2.php [hunk 1]
```diff
@@ -1110,9 +1113,13 @@ function _connect()
      */
     function _generate_identifier()
     {
-        $identifier = 'SSH-2.0-phpseclib_0.3';
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: Is this wrong on the 1.0 branch too?

Comment (bantu) [inline phpseclib/Net/SSH2.php]: Yes.

**Comment by terrafrost** [comment]
Looks good, with the caveat of what I mentioned. Maybe composer.json ought to be updated with libsodium as an entry in `suggest`?

**Comment by bantu** [comment]
@terrafrost Updated. Changed PR description.

**Comment by terrafrost** [comment]
Looks good to me!

**Pull request merged**